source/components/hardware/hwxfsleep.c

```diff
@@ -180,13 +180,29 @@ AcpiHwSleepDispatch (
 #define ACPI_WAKE_PREP_FUNCTION_ID      1
 #define ACPI_WAKE_FUNCTION_ID           2
 
+/*
+ * Some compilers can handle designated initializers, which is needed
+ * under Linux kernel builds for structures that are entirely function
+ * pointers.
+ */
+#ifdef __GNUC__
+#define ACPI_SLEEP_FUNCTION_INIT(legacy, extended) \
+              { .legacy_function   = legacy,     \
+                .extended_function = extended }
```

Comment:
I'll try your approach to making a general purpose initializer, thanks!

As to the definition order: this shouldn't matter, right? The structures is not shared outside of the kernel under Linux, yes?